cpp/src/filling/repeat.cu

```diff
@@ -72,36 +72,24 @@ struct count_accessor {
   }
 };
 
-struct compute_offsets {
-  cudf::column_view const* p_column = nullptr;
+struct count_checker {
+  cudf::column_view const& count;
 
   template <typename T>
-  std::enable_if_t<std::is_integral<T>::value, rmm::device_vector<cudf::size_type>> operator()(
-    bool check_count, rmm::cuda_stream_view stream)
+  std::enable_if_t<std::is_integral<T>::value, void> operator()(rmm::cuda_stream_view stream)
   {
     // static_cast is necessary due to bool
-    if (check_count && static_cast<int64_t>(std::numeric_limits<T>::max()) >
-                         std::numeric_limits<cudf::size_type>::max()) {
-      auto max = thrust::reduce(p_column->begin<T>(), p_column->end<T>(), 0, thrust::maximum<T>());
+    if (static_cast<int64_t>(std::numeric_limits<cudf::size_type>::max()) >
+        std::numeric_limits<cudf::size_type>::max()) {
```

Comment:
Sorry, I'm confused. Why are you comparing `static_cast<int64_t>(some_number) > some_number`? From my view point, `std::numeric_limits<cudf::size_type>::max()` always returns a `size_type` number (int32_t). You cast it to `int64_t` and do the comparison with it?

Reply:
This was the existing check. I think it really should be a static_assert.

Reply:
Can you approve now?